File: benchmarks/tasks/error_handling/error_handling.py
# SPDX-License-Identifier: MIT-0

def divide(x, y):
    if y == 0:
        return ("Err", "division by zero")
    return ("Ok", x // y)

def show_result(r):
    if r[0] == "Err":
        return f"Err {r[1]}"
    return f"Ok {r[1]}"

print(show_result(divide(10, 0)))
print(show_result(divide(10, 2)))

File: benchmarks/tasks/error_handling/error_handling.ts
// SPDX-License-Identifier: MIT-0

type Result<T, E> = { tag: "Ok"; value: T } | { tag: "Err"; value: E };

function divide(x: number, y: number): Result<number, string> {
    if (y === 0) return { tag: "Err", value: "division by zero" };
    return { tag: "Ok", value: Math.floor(x / y) };
}

function showResult(r: Result<number, string>): string {
    return r.tag === "Err" ? `Err ${r.value}` : `Ok ${r.value}`;
}

console.log(showResult(divide(10, 0)));
console.log(showResult(divide(10, 2)));
